### PR TITLE
docs(hx-menu): implement and document vars and attrs

### PR DESCRIPTION
**Pull request by BY00565233**
## Description

`<hx-menu>` : Implement and Document variables and attributes

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF-2113

## Before you request a review for this PR:

- [x] For UI changes, did you manually test in recent versions of modern browsers (Chrome, Firefox, and Safari)?
- [ ] For UI changes, did you manually test in IE11 and legacy Edge?
- [ ] Did you add component tests for any new code?
- [ ] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [ ] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/scss/components/menu/_index.scss [hunk 1]
```diff
@@ -44,7 +44,7 @@ hx-menuitem {
   background-color: transparent;
   border-radius: 2px;
   border: none;
-  color: $cyan-900;
+  color: var(--hxMenu-hxMenuitem-color, $cyan-900);
```
Comment (100stacks) [inline src/scss/components/menu/_index.scss:47]: Please update color variable to Color Palette 2.0 specs.

File: src/scss/components/menu/_index.scss [hunk 2]
```diff
@@ -56,13 +56,13 @@ hx-menuitem {
   &:hover {
     background-color: transparent;
     border-color: transparent;
-    color: $cyan-500;
+    color: var(--hxMenu-hxMenuitem-hover-color, $cyan-500);
```
Comment (100stacks) [inline src/scss/components/menu/_index.scss:59]: Please update color variable to Color Palette 2.0 specs.

File: src/scss/components/menu/_index.scss [hunk 3]
```diff
@@ -56,13 +56,13 @@ hx-menuitem {
...
   }
 
   &:active {
     background-color: transparent;
     border-color: transparent;
-    color: $cyan-700;
+    color: var(--hxMenu-hxMenuitem-active-color, $cyan-700);
```
Comment (100stacks) [inline src/scss/components/menu/_index.scss:65]: Please update color variable to Color Palette 2.0 specs.

**Review by 100stacks — CHANGES_REQUESTED** [review]

File: src/scss/components/menu/_index.scss (hunk 1 above)
Comment (BY00565233) [reply src/scss/components/menu/_index.scss]: Changes have been done, Please review

File: src/scss/components/menu/_index.scss (hunk 2 above)
Comment (BY00565233) [reply src/scss/components/menu/_index.scss]: Changes have been done, Please review

File: src/scss/components/menu/_index.scss (hunk 3 above)
Comment (BY00565233) [reply src/scss/components/menu/_index.scss]: Changes have been done, Please review

**Review by lalithkarikelli — APPROVED** [review]
LGTM

**Review by 100stacks — APPROVED** [review]
Dev LGTM

**Pull request merged**